### PR TITLE
add experiment tracking integration to text, image classification & LLM finetuning

### PR DESCRIPTION
**Pull request by rishiraj**
Worked with @soumik12345 from WandB team to successfully integrate WandB with LLM Finetuning, text & image classification. Tests are passing. Here is a notebook I used to test: https://colab.research.google.com/drive/1m0KZzcCuKRn3Rkff32XTvdIfD45hK2Va?usp=sharing

### CODE REVIEW

**Comment by rishiraj** [comment]
Worked with @soumik12345 from WandB team to successfully integrate WandB with LLM Finetuning, text & image classification. Tests are passing. Here is a notebook I used to test: https://colab.research.google.com/drive/1m0KZzcCuKRn3Rkff32XTvdIfD45hK2Va?usp=sharing

**Review by abhishekkrthakur — CHANGES_REQUESTED** [review]
Thanks for the PR. Eventually, multiple logging should be supported. Currently, we already have tensorboard. In order to get this PR merged, we need to change how the API works. The CLI should have a `--log` argument which can take inputs like `tensorboard` and `wandb`. `--log wandb` should log to wandb and so on. if nothing is provided, no logging. please let me know if the requirements are unclear

**Comment by rishiraj** [comment]
> Thanks for the PR. Eventually, multiple logging should be supported. Currently, we already have tensorboard. In order to get this PR merged, we need to change how the API works. The CLI should have a `--log` argument which can take inputs like `tensorboard` and `wandb`. `--log wandb` should log to wandb and so on. if nothing is provided, no logging. please let me know if the requirements are unclear

The requirements are clear. I am working to make the required changes.

**Comment by rishiraj** [comment]
> Thanks for the PR. Eventually, multiple logging should be supported. Currently, we already have tensorboard. In order to get this PR merged, we need to change how the API works. The CLI should have a `--log` argument which can take inputs like `tensorboard` and `wandb`. `--log wandb` should log to wandb and so on. if nothing is provided, no logging. please let me know if the requirements are unclear

@abhishekkrthakur Made the changes you requested and successfully integrated all experiment trackers like "azure_ml", "clearml", "codecarbon", "comet_ml", "dagshub", "flyte", "mlflow", "neptune", "tensorboard", and "wandb" with LLM Finetuning, text & image classification via the `--log` argument. Tests are passing. Here is a notebook I used to test: https://colab.research.google.com/drive/1m0KZzcCuKRn3Rkff32XTvdIfD45hK2Va?usp=sharing

File: src/autotrain/trainers/lm_trainer.py
Comment (abhishekkrthakur) [inline src/autotrain/trainers/lm_trainer.py:1]: remove changes from this file please. its deprecated and will be removed soon.

File: src/autotrain/trainers/text_classification.py
Comment (abhishekkrthakur) [inline src/autotrain/trainers/text_classification.py:1]: remove changes from this file please. its deprecated and will be removed soon.

**Review by abhishekkrthakur — CHANGES_REQUESTED** [review]

**Comment by abhishekkrthakur** [comment]
Please also run: `make style && make quality`

File: src/autotrain/trainers/lm_trainer.py
Comment (rishiraj) [reply src/autotrain/trainers/lm_trainer.py:1]: removed changes from this file and tested ok using the colab link.

File: src/autotrain/trainers/text_classification.py
Comment (rishiraj) [reply src/autotrain/trainers/text_classification.py:1]: removed changes from this file and tested ok using the colab link.

**Pull request merged**